### DISCUSSION TITLE
Can i call useContractRead with key value arguments?

### DISCUSSION THREAD

**Discussion by chillx2land** [wevm/wagmi#1082] [Q&A] [answered]
I feel calling with array is anxiety about array item index.

**Comment by tmm** [comment, marked as answer]
We just released automatic type-safety for contract arguments so hopefully that quells some anxiety!

https://twitter.com/wagmi_sh/status/1580591788611338240

The way wagmi represents arguments is a reflection of how they are structured in contracts. For example, if a contract function uses positional arguments:

```sol
function foo(address bar, string baz) public view returns (string)
```

`args` looks like:

```ts
args: ['0x…', 'hello world'],
// ^? (property) args: readonly [`0x${string}`, string]
```

Similarly, if your contract function has a struct argument, wagmi is able to infer the correct types for it:

```sol
struct MyStruct {
  address bar;
  string baz;
}
function foo(MyStruct myStruct) public view returns (string)
```

So you can pass in an object with named parameters:

```ts
args: [{ bar: '0x…', baz: 'hello world' }],
// ^? (property) args: readonly [{ bar: `0x${string}`, baz: string }]
```

**Reply by chillx2land** [reply]
Thanks for the detailed answer and nice new feature!
